### ISSUE TITLE
Feature Request: Video Discovery

### ISSUE THREAD

**Issue by rayblair06** [nuxt-modules/sitemap#299]
Hey 👋

### The pitch
When working with assets from external sources like a headless CMS, organising these assets and manipulating the data for image and video sitemaps can be challenging. We currently have a modular application that provides a lot of challenges when gathering this data, as the assets can be nested many items deep, so traversing and extracting that data can be quite labour intensive. 

Image discovery efficiently identifies images in pre-rendered content and provides a great and easy way to extract that data for use in sitemaps. It would be fantastic to have a similar feature for video assets.

### The solution
Within `<video>` elements, we could scrape `content_loc` and `thumbnail_loc` fields from video element sources and `poster` for sitemaps. There are complexities with `<video>` sources, whether they’re directly within the `<video>` element as `src` or as child `<source>` elements but these now be solved.

### Caveats
However, video sitemaps also require `title` and `description` fields, which `<video>` elements do not provide any solution for. One solution could be adding this metadata as `data-title` and `data-description`.

[Google's Video Sitemap guidelines](https://developers.google.com/search/docs/crawling-indexing/sitemaps/video-sitemaps) 

For example:
```
<video
    controls
    poster="https://archive.org/download/DuckAndCover_185/__ia_thumb.jpg"
    width="620"
    data-title="Duck and Cover"
    data-description="This film, a combination of animated cartoon and live action, shows young children what to do in case of an atomic attack."
>
    <source
        src="https://archive.org/download/DuckAndCover_185/CivilDefenseFilm-DuckAndCoverColdWarNuclearPropaganda_512kb.mp4"
        type="video/mp4"
    />
    <source
        src="https://archive.org/download/DuckAndCover_185/CivilDefenseFilm-DuckAndCoverColdWarNuclearPropaganda.avi"
        type="video/x-msvideo"
    />
    Sorry, your browser doesn't support embedded videos. However, you can
    <a href="https://archive.org/details/DuckAndCover_185">download it</a>
    and watch it with your favorite video player!
</video>

```

Would be 'discovered' as:

```
<video:video>
    <video:thumbnail_loc>https://archive.org/download/DuckAndCover_185/__ia_thumb.jpg</video:thumbnail_loc>
    <video:title>Duck and Cover</video:title>
    <video:description>
        This film, a combination of animated cartoon and live action, shows young children what to do in case of an atomic attack.
    </video:description>
    <video:content_loc>
        https://archive.org/download/DuckAndCover_185/CivilDefenseFilm-DuckAndCoverColdWarNuclearPropaganda_512kb.mp4
    </video:content_loc>
</video:video>

<video:video>
    <video:thumbnail_loc>https://archive.org/download/DuckAndCover_185/__ia_thumb.jpg</video:thumbnail_loc>
    <video:title>Duck and Cover</video:title>
    <video:description>
        This film, a combination of animated cartoon and live action, shows young children what to do in case of an atomic attack.
    </video:description>
    <video:content_loc>
        https://archive.org/download/DuckAndCover_185/CivilDefenseFilm-DuckAndCoverColdWarNuclearPropaganda.avi
    </video:content_loc>
</video:video>
```

I would love to know your thoughts and can help provide implementations.

**Comment by harlan-zw** [comment]
I think this is a great idea. A pr would be amazing, otherwise I can take a look when I have a chance.

**Comment by harlan-zw** [comment]
Available in [v5.3.0](https://github.com/nuxt-modules/sitemap/releases/tag/v5.3.0), thanks again!
